FirebaseRemoteConfigSwift/Sources/Codable.swift

```diff
@@ -15,9 +15,26 @@
  */
 
 import Foundation
-import FirebaseRemoteConfig
+#if SWIFT_PACKAGE
+  @_exported import FirebaseRemoteConfigInternal
+#endif // SWIFT_PACKAGE
 import FirebaseSharedSwift
 
+#if SWIFT_PACKAGE
+  // This is a trick to force generate a `FirebaseRemoteConfig-Swift.h` header
+  // that re-exports `FirebaseRemoteConfigInternal` for Objective-C clients. It
+  // is important for the below code to reference a Remote Config symbol defined
+  // in Objective-C as that will import the symbol's module
+  // (`FirebaseRemoteConfigInternal`) in the generated header. This allows
+  // Objective-C clients to import Remote Config's Objective-C API using
+  // `@import FirebaseRemoteConfig;`. This API is not needed for Swift clients
+  // and is therefore hidden for them for the foreseeable future.
+  @available(iOS 100, *)
+  @objc public extension RemoteConfig {
+    var __do_not_call: String { "" }
+  }
+#endif // SWIFT_PACKAGE
```

Comment:
This is the only "tricky" thing that needed to be done. As the comment explains, this is only for Objective-C clients.

Reply:
Will also need to add more platform specifiers... `tvOS 100, ...`